homeassistant/components/unifi/sensor.py

```diff
@@ -3,183 +3,278 @@
+        default_name=client.name or client.hostname,
+    )
+
+
+@dataclass
+class UnifiEntityLoader(Generic[_HandlerT, _DataT]):
+    """Validate and load entities from different UniFi handlers."""
+
+    allowed_fn: Callable[[UniFiController, str], bool]
+    api_handler_fn: Callable[[aiounifi.Controller], _HandlerT]
+    available_fn: Callable[[UniFiController, str], bool]
+    device_info_fn: Callable[[aiounifi.Controller, str], DeviceInfo]
+    name_fn: Callable[[_DataT], str | None]
+    object_fn: Callable[[aiounifi.Controller, str], _DataT]
+    supported_fn: Callable[[UniFiController, str], bool | None]
+    unique_id_fn: Callable[[str], str]
+    value_fn: Callable[[UniFiController, _DataT], datetime | float]
+
+
+@dataclass
+class UnifiEntityDescription(
+    SensorEntityDescription, UnifiEntityLoader[_HandlerT, _DataT]
+):
+    """Class describing UniFi sensor entity."""
+
+
+ENTITY_DESCRIPTIONS: tuple[UnifiEntityDescription, ...] = (
+    UnifiEntityDescription[Clients, Client](
+        key="Bandwidth sensor RX",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+        has_entity_name=True,
+        allowed_fn=lambda controller, _: controller.option_allow_bandwidth_sensors,
+        api_handler_fn=lambda api: api.clients,
+        available_fn=lambda controller, _: controller.available,
+        device_info_fn=async_client_device_info_fn,
+        name_fn=lambda _: "RX",
+        object_fn=lambda api, obj_id: api.clients[obj_id],
+        supported_fn=lambda controller, _: controller.option_allow_bandwidth_sensors,
+        unique_id_fn=lambda obj_id: f"rx-{obj_id}",
+        value_fn=async_client_rx_value_fn,
+    ),
+    UnifiEntityDescription[Clients, Client](
+        key="Bandwidth sensor TX",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+        has_entity_name=True,
+        allowed_fn=lambda controller, _: controller.option_allow_bandwidth_sensors,
+        api_handler_fn=lambda api: api.clients,
+        available_fn=lambda controller, _: controller.available,
+        device_info_fn=async_client_device_info_fn,
+        name_fn=lambda _: "TX",
+        object_fn=lambda api, obj_id: api.clients[obj_id],
+        supported_fn=lambda controller, _: controller.option_allow_bandwidth_sensors,
+        unique_id_fn=lambda obj_id: f"tx-{obj_id}",
+        value_fn=async_client_tx_value_fn,
+    ),
+    UnifiEntityDescription[Clients, Client](
+        key="Uptime sensor",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        has_entity_name=True,
+        allowed_fn=lambda controller, _: controller.option_allow_uptime_sensors,
+        api_handler_fn=lambda api: api.clients,
+        available_fn=lambda controller, obj_id: controller.available,
+        device_info_fn=async_client_device_info_fn,
+        name_fn=lambda client: "Uptime",
```

Comment:
Does `name_fn` always just return a string? If so, you could simply list it as such in `UnifiEntityLoader`.

Reply:
There are cases where name can be `None` just not in this platform right now.